lib/logstash/inputs/beats.rb

```diff
@@ -58,9 +59,12 @@ class LogStash::Inputs::Beats < LogStash::Inputs::Base
 
   include LogStash::PluginMixins::EventSupport::EventFactoryAdapter
 
+  include LogStash::PluginMixins::PluginFactorySupport
+
   config_name "beats"
 
-  default :codec, "plain"
+  DEFAULT_CODEC_SENTINEL = Object.new
+  default :codec, DEFAULT_CODEC_SENTINEL
```

Comment:
Applying default codec sentinel crashes the plugin initialization in the core. [Provided details with the comment](https://github.com/logstash-plugins/logstash-input-beats/pull/464#discussion_r1057624304).

Reply:
Maybe possible solution align on user's codec and enrich configs as tried with [this commit](https://github.com/logstash-plugins/logstash-input-beats/pull/464/commits/61cdf2204750d23671c0dbb663467abec017f85e) but extra codec initialization will happen if logic needs to re-initialize the codec.